src/plugins/intel_gpu/src/graph/impls/cpu/assign.cpp

```diff
@@ -45,22 +46,24 @@ struct assign_impl : public typed_primitive_impl<assign> {
     }
 
     event::ptr execute_impl(const std::vector<event::ptr>& events, assign_inst& instance) override {
+        for (auto e : events) {
+            e->wait();
```

Comment:
Wasn't this affecting the performance?

Reply:
Updated that event is returned as is, if assign can be optimized as discussed